### ISSUE TITLE
netdata/collectors/proc.plugin: wrong output for nvme disk stats, nvme disks not autodiscovered

### ISSUE THREAD

**Issue by kralewitz** [netdata/netdata#5744]
##### Bug report summary

Netdata has issues with collecting stats from nvme drives on one of my servers. On a system with 8 nvme drives, only one drive gets discovered, and even for this drive, the values produced are wrong (read/write rates are steady 8TB/s, utilization is shown to be 1 billion percent, the backlog has value of 28000 minutes, etc.). Charts for all other disks on the system are okay.

EDIT: I have the same issue on a second server, although they are identical - same OS, same HW config, so nothing really surprising here.

See the attached gist for the list of drives and contents of `/proc/diskstats`:

https://gist.github.com/kralewitz/7ab7da4f3f2ad8163d598b7c0794858d

Steps I have tried:

1) compile netdata with debug enabled and flags `0x0000000000010000 (D_PROCFILE)`

After running in debug mode, `debug.log` has nothing interesting, but `error.log` is getting spammed with this:

```
2019-03-29 18:37:05: netdata INFO  : PLUGIN[proc] : INTERNAL CHECK: host '<snip>', chart 'disk_util.nvme3c33n1' is collected in the same interpolation point: short by 335031 microseconds
```
Repeated over and over again for all charts of the nvme disk in question.

2) Adding configuration for other nvme disks to `netdata.conf` by hand, e.g.:

`
[plugin:proc:/proc/diskstats:nvme1c33n1]
enable = yes
...
`
However, the dashboard still shows only charts for the one nvme disk (`nvme3c33n1`) that was autodiscovered before. There's no mention of the other disk i configured manually in `error.log`

Any suggestions on what to try next/how to debug this further?

##### OS / Environment

Debian 9.8

`Linux <snip> 4.19.0-0.bpo.2-amd64 #1 SMP Debian 4.19.16-1~bpo9+1 (2019-02-07) x86_64 GNU/Linux`

##### Netdata version (ouput of `netdata -V`)

Both netdata v1.13.0, and the latest netdata v1.13.0-50-gb9b96490 compiled in debug mode

##### Component Name

netdata/collectors/proc.plugin

##### Steps To Reproduce

Not sure

##### Expected behavior

Netdata correctly discovers all nvme disks in use and produces meaningful charts for them :)

**Comment by kralewitz** [comment]
Also, for your reading pleasure, here's output of `/proc/diskstats` read twice with 1 second delay to better show changes in the values. Labels were taken from https://www.mjmwired.net/kernel/Documentation/iostats.txt

```
9 /dev/nvme3c33n1 [0:0]                                                                                                                                                                                                    
reads_cmpl          , first: 4542373     , second: 4542948     , diff:        575                                                                                                                                          
reads_merged        , first: 78493932    , second: 78505391    , diff:      11459                                                                                                                                          
sectors_read        , first: 667459888   , second: 667556288   , diff:      96400                                                                                                                                          
millis_reading      , first: 2363088     , second: 2363390     , diff:        302                                                                                                                                          
writes_cmpl         , first: 8147407     , second: 8148514     , diff:       1107                                                                                                                                          
writes_merged       , first: 162461697   , second: 162486055   , diff:      24358                                                                                                                                          
sectors_written     , first: 1364350380  , second: 1364554210  , diff:     203830                                                                                                                                          
millis_writing      , first: 3560147     , second: 3560691     , diff:        544                                                                                                                                          
io_current          , first: 1           , second: 1           , diff:          0                                                                                                                                          
millis_io           , first: 4570544     , second: 4571180     , diff:        636                                                                                                                                          
weighted_millis_io  , first: 5883680     , second: 5884484     , diff:        804                                                                                                                                          
                    , first: 0           , second: 0           , diff:          0                                                                                                                                          
                    , first: 0           , second: 0           , diff:          0                                                                                                                                          
                    , first: 0           , second: 0           , diff:          0
```

**Comment by kralewitz** [comment]
Regarding the "wrong" data, I took a look today at the stats for the only autodiscovered disk (attached), and most of the charts are only increasing, might be another clue to what's wrong.

![netdata_nvme](https://user-images.githubusercontent.com/15180106/55460313-fb1d6e00-55f1-11e9-9ff6-35da252318d5.png)

**Comment by vlvkobal** [comment]
The problem is that the NVMe driver doesn't set major and minor numbers. All NVMe disks have the same numbers [0:0]. The diskstats module in proc plugin uses major and minor numbers for identifying disks. It should use the devfs device name though to identify an NVMe disk correctly.

**Comment by vlvkobal** [comment]
@kralewitz please post here the contents of the `/proc/self/mountinfo`.

**Comment by kralewitz** [comment]
Here you go, the nvme disks are part of `/dev/md1`


```
20 25 0:19 / /sys rw,nosuid,nodev,noexec,relatime shared:7 - sysfs sysfs rw
21 25 0:4 / /proc rw,nosuid,nodev,noexec,relatime shared:12 - proc proc rw
22 25 0:6 / /dev rw,nosuid,relatime shared:2 - devtmpfs udev rw,size=131993352k,nr_inodes=32998338,mode=755
23 22 0:20 / /dev/pts rw,nosuid,noexec,relatime shared:3 - devpts devpts rw,gid=5,mode=620,ptmxmode=000
24 25 0:21 / /run rw,nosuid,noexec,relatime shared:5 - tmpfs tmpfs rw,size=26401632k,mode=755
25 0 253:0 / / rw,relatime shared:1 - xfs /dev/mapper/vg.system-lv.root rw,attr2,inode64,noquota
26 20 0:7 / /sys/kernel/security rw,nosuid,nodev,noexec,relatime shared:8 - securityfs securityfs rw
27 22 0:22 / /dev/shm rw,nosuid,nodev shared:4 - tmpfs tmpfs rw
28 24 0:23 / /run/lock rw,nosuid,nodev,noexec,relatime shared:6 - tmpfs tmpfs rw,size=5120k
29 20 0:24 / /sys/fs/cgroup ro,nosuid,nodev,noexec shared:9 - tmpfs tmpfs ro,mode=755
30 29 0:25 / /sys/fs/cgroup/systemd rw,nosuid,nodev,noexec,relatime shared:10 - cgroup cgroup rw,xattr,release_agent=/lib/systemd/systemd-cgroups-agent,name=systemd
31 20 0:26 / /sys/fs/pstore rw,nosuid,nodev,noexec,relatime shared:11 - pstore pstore rw
32 29 0:27 / /sys/fs/cgroup/memory rw,nosuid,nodev,noexec,relatime shared:13 - cgroup cgroup rw,memory
33 29 0:28 / /sys/fs/cgroup/pids rw,nosuid,nodev,noexec,relatime shared:14 - cgroup cgroup rw,pids
34 29 0:29 / /sys/fs/cgroup/cpu,cpuacct rw,nosuid,nodev,noexec,relatime shared:15 - cgroup cgroup rw,cpu,cpuacct
35 29 0:30 / /sys/fs/cgroup/perf_event rw,nosuid,nodev,noexec,relatime shared:16 - cgroup cgroup rw,perf_event
36 29 0:31 / /sys/fs/cgroup/devices rw,nosuid,nodev,noexec,relatime shared:17 - cgroup cgroup rw,devices
37 29 0:32 / /sys/fs/cgroup/blkio rw,nosuid,nodev,noexec,relatime shared:18 - cgroup cgroup rw,blkio
38 29 0:33 / /sys/fs/cgroup/freezer rw,nosuid,nodev,noexec,relatime shared:19 - cgroup cgroup rw,freezer
39 29 0:34 / /sys/fs/cgroup/net_cls,net_prio rw,nosuid,nodev,noexec,relatime shared:20 - cgroup cgroup rw,net_cls,net_prio
40 29 0:35 / /sys/fs/cgroup/cpuset rw,nosuid,nodev,noexec,relatime shared:21 - cgroup cgroup rw,cpuset
41 21 0:36 / /proc/sys/fs/binfmt_misc rw,relatime shared:22 - autofs systemd-1 rw,fd=38,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=11641
42 22 0:37 / /dev/hugepages rw,relatime shared:23 - hugetlbfs hugetlbfs rw,pagesize=2M
43 20 0:8 / /sys/kernel/debug rw,relatime shared:24 - debugfs debugfs rw
44 22 0:18 / /dev/mqueue rw,relatime shared:25 - mqueue mqueue rw
71 25 0:38 / /mnt/ramdisk rw,relatime shared:26 - tmpfs tmpfs rw,size=1048576k
73 25 0:39 / /mnt/cache rw,relatime shared:27 - tmpfs tmpfs rw,size=4194304k
76 71 9:1 / /mnt/ramdisk/timeshift rw,noatime,nodiratime shared:28 - xfs /dev/md1 rw,attr2,inode64,sunit=1024,swidth=7168,noquota
127 43 0:11 / /sys/kernel/debug/tracing rw,relatime shared:77 - tracefs tracefs rw
130 20 0:42 / /sys/kernel/config rw,relatime shared:79 - configfs configfs rw
143 24 0:44 / /run/user/1000 rw,nosuid,nodev,relatime shared:89 - tmpfs tmpfs rw,size=26401628k,mode=700,uid=1000,gid=100
211 41 0:45 / /proc/sys/fs/binfmt_misc rw,relatime shared:155 - binfmt_misc binfmt_misc rw
```

**Comment by vlvkobal** [comment]
@kralewitz could you test #5799? It should fix the issue with NVMe disks.

**Comment by kralewitz** [comment]
Yup, building and running netdata from `9a957f` fixes the issue - all NVMe disks get discovered correctly and their values seem sane.

Thanks for the fix!

**Comment by mfundul** [comment]
This does not seem a normal situation, no major or minor numbers. It looks like some kind of bug with the nvme driver. Relevant [link](https://answers.launchpad.net/ubuntu/+question/679738).

**Comment by vlvkobal** [comment]
@kralewitz is it possible for you to mount an NVMe disk directly and show us the information for the disk in `diskstats`, `mountinfo`, and IDs in `/dev/disk/by-id`?

**Comment by mfundul** [comment]
> 
> 
> @kralewitz is it possible for you to mount an NVMe disk directly and show us the information for the disk in `diskstats`, `mountinfo`, and IDs in `/dev/disk/by-id`?

Please make it so the that disk in question has the `c33` string in its name. The normal nvme names such as `nvme`n1` don't have any problems.

**Comment by kralewitz** [comment]
Unfortunately both servers are running production services since this morning, so I can't play with the disks anymore, but I'll have one more server available for testing, probably early next week. I will let you know when it's up.

Sorry for the delay.

**Comment by kralewitz** [comment]
Finally could play around with the new servers today, it took a bit longer than I expected, so sorry for the delay.

I created a filesystem on one of the nvme disks and mounted it:

```
root@cdn3-ttc:/# grep nvme3 /proc/diskstats 
   0       0 nvme3c33n1 682 0 16148 59 67891 258573 17174264 58210 0 584 36052 0 0 0 0
 259      13 nvme3n1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0


root@cdn3-ttc:/# grep nvme3 /proc/self/mountinfo 
122 25 259:13 / /mnt rw,relatime shared:74 - ext4 /dev/nvme3n1 rw

root@cdn3-ttc:/# ls -lh /dev/disk/by-id | grep nvme3
lrwxrwxrwx 1 root root 13 Apr 16 07:22 nvme-SAMSUNG_MZWLL1T6HAJQ-00005_S4C9NE0M200248 -> ../../nvme3n1
lrwxrwxrwx 1 root root 13 Apr 16 07:22 nvme-eui.344339304d2002480025384500000004 -> ../../nvme3n1
```
Out of `nvme3`, `nvme3n1`, and `nvme3c33n1`, only `nvme3n1` can be mounted. The other two cannot:

```
mount:  /dev/nvme3 is not a block device

mount: special device /dev/nvme3c33n1 does not exist
```

**Comment by mfundul** [comment]
Can you do a `ls -l /dev/nvme3c33n1` or locate it if it's not there?

**Comment by kralewitz** [comment]
It does not show up under `/dev`, though I see it in `/sys/class/block`:


```
# ls -lh /dev | grep nvme3
crw------- 1 root root    246,   3 Apr 16 10:23 nvme3
brw-rw---- 1 root disk    259,  14 Apr 16 10:23 nvme3n1

# ls -lh /sys/class/block/ | grep nvme3
lrwxrwxrwx 1 root root 0 Apr 16 10:44 nvme3c33n1 -> ../../devices/pci0000:00/0000:00:03.3/0000:09:00.0/nvme/nvme2/nvme3c33n1
lrwxrwxrwx 1 root root 0 Apr 16 10:44 nvme3n1 -> ../../devices/virtual/nvme-subsystem/nvme-subsys3/nvme3n1

```

The driver in use is:

```
# modinfo nvme
filename:       /lib/modules/4.19.0-0.bpo.4-amd64/kernel/drivers/nvme/host/nvme.ko
version:        1.0
license:        GPL
author:         Matthew Wilcox <willy@linux.intel.com>
```

**Comment by mfundul** [comment]
Can you also do a `lsblk --all` please?

**Comment by kralewitz** [comment]
```
# lsblk --all

sda                       8:0    0 223.6G  0 disk  
|-sda1                    8:1    0     9M  0 part  
`-sda2                    8:2    0 223.6G  0 part  
    `-vg.system-lv.root 253:0    0 134.1G  0 lvm   /
sdb                       8:16   0 223.6G  0 disk  
|-sdb1                    8:17   0     9M  0 part  
`-sdb2                    8:18   0 223.6G  0 part  
    `-vg.system-lv.root 253:0    0 134.1G  0 lvm   /
nvme2n1                 259:1    0   1.5T  0 disk  
nvme1n1                 259:7    0   1.5T  0 disk  
nvme0n1                 259:8    0   1.5T  0 disk  
nvme4n1                 259:11   0   1.5T  0 disk  
nvme6n1                 259:12   0   1.5T  0 disk  
nvme7n1                 259:13   0   1.5T  0 disk  
nvme5n1                 259:14   0   1.5T  0 disk  
nvme3n1                 259:15   0   1.5T  0 disk
```

**Comment by mfundul** [comment]
Ok, so from the data we have so far, it looks to me like there is a mapping from something that looks like a _physical_ NVMe instance (`nvme1c33n1`) to a _virtual_ NVMe instance (`nvme1n1`).

It appears that the _virtual_ instance has proper major and minor numbers, resides in `/dev/`, is mountable and usable in general.

It appears that the _physical_ instance is not generally accessible by linux applications.

Now, the problem seems to be that disk statistics are gathered for the _physical_ NVMe but not at all for the _virtual_ NVMe. This is the first time we come across any devices in diskstats that don't have any major and minor numbers and are generally inaccessible. Moreover, it is unfortunate that both the _physical_ and _virtual_ instances reside in `/proc/diskstats` even when the _virtual_ instance's statistics are all zeroes.

The ideal solution would be to be able to identify the mapping, allowing us to synthesize useful information from the 2 sources. Ideally we should be able to get major and minor numbers from the _virtual_ instance and statistics from the _physical_ instance.

It still looks like a kernel bug to me (regarding `/proc/diskstats` entries), any additional input will be greatly appreciated.

**Comment by mfundul** [comment]
In kernel 4.19 this line `https://elixir.bootlin.com/linux/v4.19/source/block/genhd.c#L1341` of function `diskstats_show()` produces zeroed major and minor numbers.

**Comment by kralewitz** [comment]
Update: we now have another server with NVMe disks, slightly different model though, and newer kernel. On the new machine, `/proc/diskstats` shows only the raw device names without the embedded controller ID, with the correct device numbers and stats, so netdata collects the metrics correctly there.

I will try to schedule a kernel upgrade on one of the original machines and let you know if the problem is indeed resolved on the newer kernel.

original:
```
Linux <snip> 4.19.0-0.bpo.2-amd64 #1 SMP Debian 4.19.16-1~bpo9+1 (2019-02-07) x86_64 GNU/Linux

Node             SN                   Model                                    Namespace Usage                      Format           FW Rev  
---------------- -------------------- ---------------------------------------- --------- -------------------------- ---------------- --------
/dev/nvme0n1     <snip>      SAMSUNG MZWLL1T6HAJQ-00005               1           1.60  TB /   1.60  TB    512   B +  0 B   GPJA0B3Q
```


new:
```
Linux <snip> 4.19.0-0.bpo.5-amd64 #1 SMP Debian 4.19.37-3~bpo9+1 (2019-05-18) x86_64 GNU/Linux


Node             SN                   Model                                    Namespace Usage                      Format           FW Rev  
---------------- -------------------- ---------------------------------------- --------- -------------------------- ---------------- --------
/dev/nvme0n1     <snip>       SAMSUNG MZQLB7T6HMLA-00007               1           1.28  TB /   7.68  TB    512   B +  0 B   EDB5002Q
```

**Comment by vlvkobal** [comment]
@kralewitz have you got any updates on the issue so far?

**Comment by kralewitz** [comment]
My maintenance window for the upgrade got pushed back, so not yet. Sorry :( I'll try to schedule it for this week, not sure it will get approved though.

**Comment by xPaw** [comment]
I just upgraded from debian 9 to debian 10, as a result the kernel got updated to 4.19 and after rebooting I am seeing constantly high disk utilization in netdata, but iotop and other utilities do not show anything out of the ordinary.

We're running nvme disks as well.

This munin issue might also be related to this: https://github.com/munin-monitoring/munin/issues/1119

**Comment by vlvkobal** [comment]
@xPaw Please provide the output of
```
for i in `seq 60`; do grep "0 nvme1n1" /proc/diskstats; sleep 1; done
```
while the utilization chart is close to 100%.

**Comment by xPaw** [comment]
It's been pretty constantly close to 100% ever since I rebooted.

Here's the output:
```
$ for i in `seq 60`; do grep "0 nvme1n1" /proc/diskstats; sleep 1; done
 259       0 nvme1n1 231544 7 8254881 83306 713011 263473 18267876 1060882 0 10721680 11021360 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 713177 263527 18270768 1061079 0 10722472 11022152 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 713291 263594 18272217 1061189 0 10723328 11023008 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 713307 263600 18272329 1061220 0 10724184 11023864 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 713482 263649 18275758 1061384 0 10725096 11024776 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 713652 263691 18278703 1061603 0 10726004 11025696 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 713686 263707 18279463 1061644 0 10726936 11026628 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 713706 263716 18281637 1061678 0 10727856 11027548 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 713718 263721 18281733 1061699 0 10728824 11028516 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 713767 263739 18282452 1061757 0 10729764 11029456 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 713801 263749 18282808 1061805 0 10730692 11030384 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 713817 263754 18282936 1061831 0 10731632 11031324 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 713842 263764 18283131 1061871 0 10732612 11032304 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 713983 263803 18285776 1062026 0 10733564 11033256 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 714054 263824 18286686 1062080 0 10734548 11034240 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 714066 263827 18286766 1062098 0 10734908 11034600 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 714096 263837 18287102 1062138 0 10735424 11035116 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 714119 263844 18289265 1062182 0 10736424 11036116 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 714119 263844 18289265 1062182 0 10736424 11036116 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 714127 263846 18289313 1062202 0 10737044 11036736 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 714188 263862 18290186 1062279 0 10737700 11037392 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 714194 263865 18290238 1062292 0 10738532 11038224 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 714234 263882 18290735 1062348 0 10739064 11038756 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 714299 263898 18291645 1062428 0 10739804 11039496 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 714301 263898 18291649 1062434 0 10739916 11039608 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 714343 263909 18292172 1062489 0 10740464 11040156 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 714355 263913 18292260 1062512 0 10741216 11040908 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 714421 263933 18295078 1062592 0 10742000 11041692 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 714446 263941 18295369 1062631 0 10743000 11042692 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 714510 263959 18296340 1062725 0 10743932 11043624 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 714544 263967 18296680 1062769 0 10744760 11044452 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 714556 263971 18296768 1062792 0 10745444 11045136 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 714556 263971 18296768 1062792 0 10745444 11045136 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 714573 263976 18296891 1062823 0 10746104 11045796 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 714611 263986 18297370 1062858 0 10746940 11046632 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 714733 264021 18299252 1063003 0 10747856 11047548 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 714743 264027 18299360 1063020 0 10748768 11048460 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 714798 264043 18302185 1063085 0 10749360 11049052 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 714894 264067 18304064 1063198 0 10750332 11050024 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 714907 264071 18304155 1063224 0 10751092 11050784 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 714979 264092 18305026 1063307 0 10751956 11051648 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 715044 264114 18306240 1063370 0 10752960 11052652 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 715118 264131 18307461 1063458 0 10753544 11053236 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 715171 264146 18308412 1063518 0 10754364 11054060 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 715243 264165 18309264 1063604 0 10755196 11054892 0 0 0 0
 259       0 nvme1n1 231544 7 8254881 83306 715303 264180 18310067 1063676 0 10755908 11055604 0 0 0 0
 259       0 nvme1n1 231549 7 8255041 83306 715351 264194 18310547 1063729 0 10756732 11056428 0 0 0 0
 259       0 nvme1n1 231549 7 8255041 83306 715411 264228 18313267 1063807 0 10757640 11057336 0 0 0 0
 259       0 nvme1n1 231549 7 8255041 83306 715565 264263 18315826 1063994 0 10758552 11058252 0 0 0 0
 259       0 nvme1n1 231549 7 8255041 83306 715579 264268 18315942 1064016 0 10759336 11059036 0 0 0 0
 259       0 nvme1n1 231549 7 8255041 83306 715632 264282 18316646 1064085 0 10760124 11059824 0 0 0 0
 259       0 nvme1n1 231549 7 8255041 83306 715689 264323 18317449 1064145 0 10760916 11060616 0 0 0 0
 259       0 nvme1n1 231549 7 8255041 83306 715793 264355 18319458 1064248 0 10761736 11061436 0 0 0 0
 259       0 nvme1n1 231549 7 8255041 83306 715806 264358 18319541 1064274 0 10762572 11062272 0 0 0 0
 259       0 nvme1n1 231549 7 8255041 83306 715828 264368 18319720 1064310 0 10763444 11063144 0 0 0 0
 259       0 nvme1n1 231549 7 8255041 83306 715867 264375 18320191 1064331 0 10764004 11063704 0 0 0 0
 259       0 nvme1n1 231549 7 8255041 83306 715963 264402 18321549 1064439 0 10765008 11064708 0 0 0 0
 259       0 nvme1n1 231549 7 8255041 83306 716028 264420 18324354 1064526 0 10765924 11065624 0 0 0 0
 259       0 nvme1n1 231549 7 8255041 83306 716089 264462 18346096 1064731 0 10766872 11066612 0 0 0 0
 259       0 nvme1n1 231549 7 8255041 83306 716091 264462 18346100 1064737 0 10767168 11066908 0 0 0 0
```

**Comment by vlvkobal** [comment]
You have a series `11021360 11022152 11023008 11023864 11024776 11025696 11026628 ...` for the # of milliseconds spent doing I/Os, which means you have `792 856 856 912 908 932 ...` ms spent in every iteration. This is quite close to 100%. Thus, Netdata displays correct data.

Please read the chart description
> Disk Utilization measures the amount of time the disk was busy with something. This is not related to its performance. 100% means that the system always had an outstanding operation on the disk. Keep in mind that depending on the underlying technology of the disk, 100% here may or may not be an indication of congestion.

The new driver might work in a different way and that can be the cause of different values in the disk utilization chart.

**Comment by xPaw** [comment]
I'm not entirely sure why that would be the case though, it also keeps the disk backlog chart consistently around 100. Stopping services that hit the disk (mysql) doesn't really make that go down to zero as you would expect. Iotop reports nothing, iostat shows high idle percentage.

Are you sure it could be the driver? Either way, this makes the netdata chart and alert unusuable, not sure how to fix that.

**Comment by mfundul** [comment]
If even `iostat` also shows 100% then it is probably a well known bug in the linux kernel that has been due to `/proc/diskstats` being re-written and optimized by the kernel developers. Look [here](https://github.com/sysstat/sysstat/issues/187) for reference.

You can run `iostat -x -m 1` on your terminal and notice your device's utilization. Netdata is very consistent with sysstat's `iostat` in this metric and there is not much that can be done, other than a fix for this regression in the kernel itself.

**Comment by mfundul** [comment]
More information:
https://unix.stackexchange.com/questions/517132/dd-is-running-at-full-speed-but-i-only-see-20-disk-utilization-why

https://unix.stackexchange.com/questions/517667/nvme-disk-shows-80-io-utilization-partitions-show-0-io-utilization?noredirect=1&lq=1

https://www.spinics.net/lists/kernel/msg3155496.html

**Comment by vlvkobal** [comment]
Incorrect major and minor numbers for nvme disks is a bug for a specific kernel (4.19). So, I'm closing the issue.

@kralewitz if you get any additional information/ideas on the bug we can help you with, feel free to reopen the issue.

**Comment by vlvkobal** [comment]
The analogous issue (#9560) is happening on Proxmox 6.2 5.4.44-2.

@csutcliff if you get any additional information/ideas, feel free to reopen the issue.

**Comment by bdossantos** [comment]
Hi! I encountered the same "bug" with some Debian 10 box backed by NVME SSD and a 4.19 kernel. 

"fun" fact: In my case, It seem switching I/O scheduler from `none` to `mq-deadline` fixed the reporting of the real disk utilization ...

```
echo "mq-deadline" | sudo tee /sys/block/nvme*/queue/scheduler
```

<img width="1146" alt="Screen Shot 2020-11-09 at 7 10 03 PM" src="https://user-images.githubusercontent.com/245284/98583819-49621b00-22c5-11eb-9947-4700229f78ba.png">

I don't have enough perspective to compare performance, but at least disk utilization seem real.

**Comment by vlvkobal** [comment]
The issue was fixed in #10841. @kralewitz please test it on your system.

**Comment by Thin-Troll** [comment]
@bdossantos, 
how did you solve this problem?
I have exactly the same conditions and situations
